### PR TITLE
Fix Qwen3.6 Olive mixed-precision export

### PR DESCRIPTION
**Pull request by titaiwangms**
## Summary

- keep Qwen3.5/3.6-MoE GatedDeltaNet projections in floating point for Olive-format checkpoints
- keep the shared-expert gate floating point while preserving quantized attention, shared-expert MLP, and fused QMoE experts
- preserve existing dense, GPTQ/AWQ, and unquantized graph construction

This aligns Mobius with the module selection introduced by microsoft/Olive#2630 and unblocks the multimodal Qwen3.6-35B-A3B recipe in microsoft/olive-recipes#588.

## Validation

- `python -m pytest -q src/mobius/models/qwen35_test.py src/mobius/components/_gated_deltanet_test.py src/mobius/components/_moe_test.py` (57 passed)
- targeted text/VL tests cover mixed initializer binding, dense Olive behavior, non-Olive graph preservation, and fused QMoE weights

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]
Could you explain why it's excluded? Is it due to accuracy reasons?

File: src/mobius/models/moe.py [hunk 1]
```diff
@@ -276,11 +276,18 @@ class Qwen2MoELayer(MoELayer):
...
     ``linear_class``, when provided, is used for the shared expert's MLP and
-    gate, and is threaded through to :class:`MoELayer` so it also drives the
-    dense loop-over-experts fallback (used when the config doesn't match the
-    native QMoE ABI). It has no effect on the fused QMoE path, whose
-    quantized parameters are constructed directly by
-    :meth:`MoELayer._init_qmoe_parameters`.
+    -- by default only -- for ``shared_expert_gate``, and is threaded through
+    to :class:`MoELayer` so it also drives the dense loop-over-experts
+    fallback (used when the config doesn't match the native QMoE ABI). It has
+    no effect on the fused QMoE path, whose quantized parameters are
```
[the remark this answers is not in the document]
Comment (titaiwangms) [reply src/mobius/models/moe.py]: Reworded in 192a8e6e to state explicitly that linear_class always drives the shared-expert MLP and drives the gate unless shared_expert_gate_class overrides it.

File: src/mobius/models/qwen35.py [hunk 2]
```diff
@@ -263,9 +320,31 @@ class Qwen35MoEBlock(Qwen2MoELayer):
...
+        config: ArchitectureConfig,
+        gate: nn.Module | None = None,
+        linear_class: type | None = None,
+    ):
+        super().__init__(
+            config,
+            gate=gate,
+            linear_class=linear_class,
+            # ``None`` falls back to ``linear_class`` (quantized gate).
+            shared_expert_gate_class=Linear if _keeps_modules_float(config) else None,
```
[the remark this answers is not in the document]
Comment (titaiwangms) [reply src/mobius/models/qwen35.py]: Fixed in 192a8e6e; the comment now describes only the fallback behavior and no longer assumes that linear_class is quantized.

**Comment by titaiwangms** [comment]
On why Olive excludes these modules: the two cases have slightly different motivations.

- `linear_attn` is a GatedDeltaNet/SSM recurrent block. Olive deliberately excludes the whole subtree because its generic 2-D Linear quantization path was not intended/validated for projections feeding the state-space recurrence; this is primarily a support/correctness boundary rather than a measured Mobius accuracy choice.
- `shared_expert_gate` is a `[1, hidden]` sigmoid gate controlling the always-active shared-expert contribution. Olive keeps routing-like signals in float, and int4 quantization is disproportionately lossy for a single-output-row gate, so accuracy is the reason there.

The Mobius change itself is a checkpoint-layout compatibility fix: post-#2630 Olive checkpoints contain float tensors for both groups. Building MatMulNBits modules for them produces packed-shape/scale initializers that do not exist, so export cannot bind the weights. The rest of the decoder remains quantized.
